docs/tutorials/dp-for-cross-silo-fl.md

```diff
@@ -0,0 +1,121 @@
+            Please refer to the following function.
+            privacy_engine.make_private(
+                module=self.model_,
+                optimizer=self.optimizer_,
+                data_loader=self.train_loader_,
+                noise_multiplier=dp_noise_multiplier,
+                max_grad_norm=dp_max_grad_norm,
+            )
+            """
+    ```
+    ```python
+    if dp:
+        if not ModuleValidator.is_valid(self.model_):
+            self.model_ = ModuleValidator.fix(self.model_)
+    ```
+- In the corresponding FL experiment [config file](../../examples/pipelines/fl_cross_silo_literal/config.yaml), add the DP relevant parameters, like below:
+    ```yaml
+     # Differential privacy
+     dp: true # Flag to enable/disable differential privacy
+     dp_target_epsilon: 50.0 # Smaller epsilon means more privacy, more noise
+     dp_target_delta: 1e-5 # The target δ of the (ϵ,δ)-differential privacy guarantee. Generally, it should be set to be less than the inverse of the size of the training dataset. 
+     dp_max_grad_norm: 1.0 # Clip per-sample gradients to this norm (DP)
+    ```
+- In the corresponding FL experiment [submit script](../../examples/pipelines/fl_cross_silo_literal/submit.py), add the DP relevant args, like below:
+    ```python
+    # Differential Privacy
+    dp=YAML_CONFIG.training_parameters.dp,
+    # DP target epsilon
+    dp_target_epsilon=YAML_CONFIG.training_parameters.dp_target_epsilon,
+    # DP target delta
+    dp_target_delta=YAML_CONFIG.training_parameters.dp_target_delta,
+    # DP max gradient norm
+    dp_max_grad_norm=YAML_CONFIG.training_parameters.dp_max_grad_norm,
+    # Total num of iterations
+    total_num_of_iterations=YAML_CONFIG.training_parameters.num_of_iterations,
+    ```
+It is also worthy to point out that, the provided DP cross-silo FL experiments in the current repo could be used for both DP-training and non-DP training. You can quickly turn DP off by setting the corresponding FL experiment `config.yaml` file's `dp` parameter to `false`, and turn DP on by setting `dp` parameter to `true`.
+
+### Tips and pitfalls about DP
+- DP privacy protection doesn't come without cost, and there is a model utility-privacy tradeoff. Epsilon is the DP privacy budget, and smaller epsilon means more privacy protection, more noises -- and hence worse model utility. For very sensitive scenarios, e.g., large language generative modeling tasks on very sensitive private data, DP might be desired to ensure approprite level of privacy protection. For less sensitive scenarios (e.g., standard classification modeling tasks), DP might be an overkill. Users should be aware of such model utility-privacy trade-off for DP, and evaluate their scenarios carefully to decide whether to apply DP, and if applying DP, with how much privacy budget.
```

Comment:
This is a long section of text, can you add some structure here?

ex:

#### DP privacy doesn't come without cost

There is a model utility tradeoff...

#### Convergence is DP models is not trivial

...

etc

Reply:
updated.